### ISSUE TITLE
Linux Compatibility with Powershell Core

### ISSUE THREAD

**Issue by colombeen** [dnewsholme/PasswordState-Management#20]
It would be awesome if the module could be compatible with powershell core... We are using more and more linux servers and at this point I can't seem to get the module to work on CentOS.

**Comment by dnewsholme** [comment]
So i've checked this. While module is core compatible it doesn't encounter issues on Linux is seems. It looks to be to do with file paths. Paths in windows use backslash but linux will only user forwardslash.

I can change them all to forwardslash as windows mostly works with forward slash. But looking at a better solution now.

Also ConvertFrom-SecureString fails on pwsh core < 6.2

**Comment by colombeen** [comment]
Sux that it's such nonsense that keeps the module from being compatible...

**Comment by dnewsholme** [comment]
I've managed on my development branch to fix the file path issue. However there is an issue with Passing credentials to the winapi endpoint.

````
Invoke-RestMethod : GSSAPI operation failed with error - An invalid status code was supplied (Configuration file does not specify default realm).
At line:1 char:1
+ Invoke-RestMethod -Uri "https://pws.linkgroup.co.uk/winapi/searchpass ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Invoke-RestMethod], Win32Exception
+ FullyQualifiedErrorId : System.ComponentModel.Win32Exception,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
````
Issue should be fixed with the move to .Net core 3.0
https://github.com/PowerShell/PowerShell/issues/7801

Related merge:
https://github.com/dotnet/corefx/pull/35383

**Comment by colombeen** [comment]
By using this before the passwordstate module we can get results, but also error messages on file paths (not using a newer version then .95): 
```powershell
$env:DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER=0
```

**Comment by dnewsholme** [comment]
> By using this before the passwordstate module we can get results, but also error messages on file paths (not using a newer version then .95):
> 
> ```powershell
> $env:DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER=0
> ```

Interestingly doing this on my ubuntu 18.04 box doesn't return any results and i get a 401 despite the correct credentials. Which i'm sure is due to the kerberos issues.

**Comment by colombeen** [comment]
Most of our systems run CentOS

**Comment by dnewsholme** [comment]
If you want to test you can pull down my develop branch which has the path fixes. I don't currently have a CentOS box available to test on.

**Comment by dnewsholme** [comment]
The code in my develop branch works when using an APIKey for Authentication but not kerberos.  As it somewhat works better on linux than it did before ill merge it with master and publish to the psgallery but ill keep this open until i can get the windows authentication working too.

**Comment by dnewsholme** [comment]
new build is up.
https://www.powershellgallery.com/packages/passwordstate-management/0.0.96

**Comment by colombeen** [comment]
Tried the new version on CentOS with "WindowsAuthOnly", works like a charm with $env:DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER set to 0

**Comment by dnewsholme** [comment]
Must be an issue with my ubuntu box then, i've added setting $env:DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER  to 0 automatically on linux machines.
 resolved in #22 

Release version : https://www.powershellgallery.com/packages/passwordstate-management/0.0.97
